bundle/config/mutator/translate_paths.go

```diff
@@ -150,55 +151,56 @@ func translateNoOp(literal, localFullPath, localRelPath, remotePath string) (str
 	return localRelPath, nil
 }
 
-type transformer struct {
-	// A directory path relative to which `path` will be transformed
-	dir string
-	// A path to transform
-	path *string
-	// Name of the config property where the path string is coming from
-	configPath string
-	// A function that performs the actual rewriting logic.
-	fn rewriteFunc
+func (m *translatePaths) rewriteValue(b *bundle.Bundle, p dyn.Path, v dyn.Value, fn rewriteFunc, dir string) (dyn.Value, error) {
+	out := v.MustString()
+	err := m.rewritePath(dir, b, &out, fn)
+	if err != nil {
+		if target := (&ErrIsNotebook{}); errors.As(err, target) {
+			return dyn.InvalidValue, fmt.Errorf(`expected a file for "%s" but got a notebook: %w`, p, target)
+		}
+		if target := (&ErrIsNotNotebook{}); errors.As(err, target) {
+			return dyn.InvalidValue, fmt.Errorf(`expected a notebook for "%s" but got a file: %w`, p, target)
+		}
+		return dyn.InvalidValue, err
+	}
+
+	return dyn.NewValue(out, v.Location()), nil
 }
 
-type transformFunc func(resource any, dir string) *transformer
+func (m *translatePaths) rewriteRelativeTo(b *bundle.Bundle, p dyn.Path, v dyn.Value, fn rewriteFunc, dirs []string) (dyn.Value, error) {
+	var err error
 
-// Apply all matches transformers for the given resource
-func (m *translatePaths) applyTransformers(funcs []transformFunc, b *bundle.Bundle, resource any, dir string) error {
-	for _, transformFn := range funcs {
-		transformer := transformFn(resource, dir)
-		if transformer == nil {
+	for _, dir := range dirs {
```

Comment:
I assume 2nd directory is the "fallback" one to have backward compatibility? Shall we show the warning in this case ?

Reply:
That's correct. I can add a TODO to emit one and make the fallback more explicit.